ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/PushRouting.java

```diff
@@ -238,13 +290,86 @@ public void close() {
     });
   }
 
+  private void checkForNewHostsOnContext(
+      final ServiceContext serviceContext,
+      final PushPhysicalPlan pushPhysicalPlan,
+      final ConfiguredStatement<Query> statement,
+      final Set<KsqlNode> hosts,
+      final LogicalSchema outputSchema,
+      final TransientQueryQueue transientQueryQueue,
+      final PushConnectionsHandle pushConnectionsHandle
+  ) {
+    pushPhysicalPlan.getContext().runOnContext(v ->
+        checkForNewHosts(serviceContext, pushPhysicalPlan, statement, outputSchema,
+            transientQueryQueue, pushConnectionsHandle));
+  }
+
+  private void checkForNewHosts(
+      final ServiceContext serviceContext,
+      final PushPhysicalPlan pushPhysicalPlan,
+      final ConfiguredStatement<Query> statement,
+      final LogicalSchema outputSchema,
+      final TransientQueryQueue transientQueryQueue,
+      final PushConnectionsHandle pushConnectionsHandle
+  ) {
+    VertxUtils.checkContext(pushPhysicalPlan.getContext());
+    if (pushConnectionsHandle.isClosed()) {
+      return;
+    }
+    final Set<KsqlNode> updatedHosts = registryToNodes.apply(
+        pushPhysicalPlan.getScalablePushRegistry());
+    final Set<KsqlNode> hosts = pushConnectionsHandle.getAllHosts();
+    final Set<KsqlNode> newHosts = Sets.difference(updatedHosts, hosts).stream()
+        .filter(node -> !node.isLocal())
+        .collect(Collectors.toSet());
+    final Set<KsqlNode> removedHosts = Sets.difference(hosts, updatedHosts);
+    if (newHosts.size() > 0) {
+      LOG.info("Dynamically adding new hosts {}", newHosts);
```

Comment:
I tried following all long with the logic of finding new hosts but can't quite find exactly where you do it, I was expecting a similar call to `getRemoteHosts()`. Or is it just tracked by `pushConnectionsHandle`?